src/v/kafka/latency_probe.h

```diff
@@ -80,8 +80,9 @@ class latency_probe {
     std::unique_ptr<hdr_hist::measurement> auto_produce_measurement() {
         return _produce_latency.auto_measure();
     }
-    std::unique_ptr<hdr_hist::measurement> auto_fetch_measurement() {
-        return _fetch_latency.auto_measure();
+
+    void record_fetch_latency(std::chrono::microseconds micros) {
+        _fetch_latency.record(micros.count());
```

Comment:
This is part of not using the `auto_measure` class for timing each fetch sub-query, since it always allocates. Just record the beginning timestamp the old-fashioned way.